os_net_config/common.py

```diff
@@ -146,6 +158,80 @@ def get_file_data(filename):
         return ''
 
 
+def ensure_directory_presence(filepath):
+    dir_path = os.path.dirname(filepath)
+    if not os.path.exists(dir_path):
+        os.makedirs(dir_path)
+
+
+def write_yaml_config(filepath, data):
+    ensure_directory_presence(filepath)
```

Comment:
Just for reference, [`os.makedirs`](https://docs.python.org/3/library/os.html#os.makedirs) has an `exist_ok=` flag, this function is not required. You can replace it with:

```python
os.makedirs(os.path.dirname(filepath), exist_ok=True)
```